pkg/config/config.go

```diff
@@ -0,0 +1,123 @@
+package config
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+)
+
+const (
+	ConfigFileName       = "weave-gitops-config.json"
+	WrongConfigFormatMsg = `Your CLI configuration for Weave GitOps should represent a JSON object format:
+	{
+		"analytics": true,
+		"userId": ""
+	}
+Please set you configuration with: gitops set config`
+)
+
+type GitOpsCLIConfig struct {
+	Analytics bool   `json:"analytics"`
+	UserID    string `json:"userId"`
+}
+
+// GetConfigPath composes the path of the config file for the GitOps CLI
+func GetConfigPath(filename string) (string, error) {
+	var err error
+
+	userConfigDir, err := os.UserConfigDir()
+	if err != nil {
+		return "", fmt.Errorf("user config directory could not be determined: %w", err)
+	}
+
+	configPath := filepath.Join(userConfigDir, filename)
+
+	return configPath, nil
+}
+
+// OpenConfigFile opens existing config file or creates one if it does not exist
+func OpenConfigFile(configPath string, shouldCreate bool) (*os.File, error) {
+	_, err := os.Stat(configPath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) || !shouldCreate {
```

Comment:
@ozamosi should the outer and inner conditions be combined on the same line? I tried it, it looks too complex.:)

Reply:
To me, the bit that makes it look complex comes from the number of negations. I'd probably find this more readable if written as:

```
if err != nil {
  if errors.Is(err, os.ErrNotExist) && shouldCreate {
    // we'll just create a new file
    err = nil
  } else {
    return nil, err
  }
}
```

Reply:
Reworked it as suggested, just had to remove the `err = nil` assignment because I was getting `::error file=pkg/config/config.go,line=49,col=4::ineffectual assignment to err (ineffassign)` from the linter.